### ISSUE TITLE
[MNG-5995] Maven itself cannot run without maven-compat

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7447]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** opened **[MNG-5995](https://issues.apache.org/jira/browse/MNG-5995?redirect=false)** and commented

For all the 3.0 versions of the maven-plugins we require to not depend on maven-compat anymore. However, the Maven distribution still requires maven-compat. A simple proof: remove `lib/maven-compat-3.x.y` and execute `mvn validate`.
You'll get the following exception:

```
[WARNING] Error injecting: org.apache.maven.project.DefaultProjectBuildingHelper
com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) No implementation for org.apache.maven.repository.RepositorySystem was bound.
  while locating org.apache.maven.project.DefaultProjectBuildingHelper
```

Reason: there's only one implementation: o.a.m.r.l.LegacyRepositorySystem, which is part of maven-compat.

---

**Affects:** 3.3.9

**Issue Links:**
- [MSITE-833](https://issues.apache.org/jira/browse/MSITE-833) Remove dependency to maven-compat

- [MNG-7071](https://issues.apache.org/jira/browse/MNG-7071) Remove warning about ignored profiles.xml

- [MRRESOURCES-92](https://issues.apache.org/jira/browse/MRRESOURCES-92) Upgrade to Maven 3.0 compatiblity

- [MENFORCER-267](https://issues.apache.org/jira/browse/MENFORCER-267) Upgrade to make Maven 3.1+

- [MNG-6642](https://issues.apache.org/jira/browse/MNG-6642) Tycho-based modules do not build with 3.6.1 (works with 3.6.0)

- [MNG-6561](https://issues.apache.org/jira/browse/MNG-6561) New RepositorySystem implementation as replacement of deprecated from maven-compat

**Remote Links:**
- [GitHub Pull Request #185
](https://github.com/apache/maven/pull/185)

2 votes, 7 watchers

**Comment by jira-importer** [comment]
**[Rajiv Jain](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rajivpj)** commented

Why do you say that?

To build the project I ran mvn -Drat.skip=true clean package and mvn -Drat.skip=true validate. -Drat.skip=true was required so that I avoid the RATException. Have I missed anything?

What else am I missing?

Below is the build output:

[INFO] RAT will not execute since it is configured to be skipped via system property rat.skip.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] Apache Maven ....................................... SUCCESS [ 0.991 s]
[INFO] Maven Model ........................................ SUCCESS [ 0.146 s]
[INFO] Maven Artifact ..................................... SUCCESS [ 0.047 s]
[INFO] Maven Plugin API ................................... SUCCESS [ 5.272 s]
[INFO] Maven Builder Support .............................. SUCCESS [ 0.021 s]
[INFO] Maven Model Builder ................................ SUCCESS [ 3.280 s]
[INFO] Maven Settings ..................................... SUCCESS [ 0.023 s]
[INFO] Maven Settings Builder ............................. SUCCESS [ 2.879 s]
[INFO] Maven Repository Metadata Model .................... SUCCESS [ 0.021 s]
[INFO] Maven Aether Provider .............................. SUCCESS [ 11.856 s]
[INFO] Maven Core ......................................... SUCCESS [ 13.564 s]
[INFO] Maven Embedder ..................................... SUCCESS [ 5.267 s]
[INFO] Apache Maven Distribution .......................... SUCCESS [ 2.977 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 46.962 s
[INFO] Finished at: 2016-07-20T21:02:34+01:00
[INFO] Final Memory: 47M/328M
[INFO] ------------------------------------------------------------------------

**Comment by jira-importer** [comment]
**[Rajiv Jain](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rajivpj)** commented

Could you please let me know what else I am missing? I think the clue is removing anything where the implemention involves the RepositorySystem?

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[Sylwester Lachiewicz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slachiewicz), as far as I understand your GH comments, this requires work in other components and cannot be fixed in 3.6.1? Can it? Maybe 3.7.0?

**Comment by jira-importer** [comment]
**[Sylwester Lachiewicz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slachiewicz)** commented

No, with my change You can just remove maven-compat.jat from dist and Maven should run fine for simple projects so it no longer depends on 2.0x classes. But.. we still have many old plugins and without compat this will not run.
So I used this version to test what plugins are necessary to compile Maven Core (ie migrate all required plugins to 3.0+ api) and others dev can also test.

So in short - It would b good to include in 3.6.1

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Right, ideally maven-compat is not available on the classpath, unless a plugin explicitly requires it. Only in that case the maven-compat jar of the Maven Runtime should be added to the classpath (like now: the version of maven dependencies of the plugin doesn't have any effect, they are filtered out and the Maven Runtime dependencies are used).
This is probably too complex is doesn't gain enough.
However, we might hit similar issues with the next major version of Maven, but that's for later.

**Comment by jira-importer** [comment]
**[Sylwester Lachiewicz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slachiewicz)** commented

Fixed in [bb3ec5da71d26d105972392f0a20bc61bc5d8c53](https://gitbox.apache.org/repos/asf?p=maven.git;a=commit;h=bb3ec5da71d26d105972392f0a20bc61bc5d8c53)
